src/formatter.rs

```diff
@@ -1,7 +1,7 @@
-pub trait FormatUserOutput<T> {
-    /// Format user output as output type `T`.
-    fn format_line(&self) -> String;
-}
+pub trait FormatUserOutput {
+    /// Format user output as human-readable string
+    fn format_human(&self) -> String;
 
-pub struct Json;
-pub struct Human;
+    /// Format user output as json string
+    fn format_json(&self) -> String;
```

Comment:
I'm not yet sure about this change: I don't particularly like added multiple output types to a single trait.